src/socket.c

```diff
@@ -742,10 +745,21 @@ void next_cmd_from_buffer(D_SOCKET *dsock)
     {
       telopt = 1;
     }
-    else if (telopt == 1 && (dsock->inbuf[i] == (signed char) DO || dsock->inbuf[i] == (signed char) DONT))
+    else if (telopt == 1
+             && (dsock->inbuf[i] == (signed char) DO
+                 || dsock->inbuf[i] == (signed char) DONT
+                 || dsock->inbuf[i] == (signed char) SB))
     {
       telopt = 2;
     }
+    else if (telopt == 1 && gmcp && dsock->inbuf[i] == (signed char) SE)
+    {
+      telopt = 0;
+      gmcp = FALSE;
+      dsock->next_command[j] = '\0';
+      gmcpReceived(dsock);
+      dsock->next_command[j = 0] = '\0';
```

Comment:
Isn't this too clever? :)

Reply:
Might be. But I think it's one of those, "once you extend it you can clean it up a bit" situations. :D

Didn't want to overthink things now since I don't know how it will be used in the future.

Reply:
I found funny doing the assignment inside array reference, never saw it before.

Reply:
Never used it before. But I felt it was nicer to keep the line count low here.

Reply:
@rogersm Wow, great eye! Definitely a nuance I didn't notice, but I'm okay with the convention here. _But_ since you brought it to my attention, I do have a note a little further down that I'll address.